test/test_little_dict.jl

```diff
@@ -0,0 +1,474 @@
+            LittleDict{Int,Float64})
+        @test isa(@inferred(LittleDict{Int,Float64}(Pair(1, 1), Pair(2, 2))), LittleDict{Int,Float64})
+        @test isa(@inferred(LittleDict(Pair(1, 1.0), Pair(2, 2.0), Pair(3, 3.0))), LittleDict{Int,Float64})
+        @test LittleDict(()) == LittleDict{Any,Any}()
+
+        @test isa(@inferred(LittleDict([Pair(1, 1.0), Pair(2, 2.0)])), LittleDict{Int,Float64})
+        @test_throws ArgumentError LittleDict([1,2,3,4])
+
+        iter = Iterators.filter(x->x.first>1, [Pair(1, 1.0), Pair(2, 2.0), Pair(3, 3.0)])
+        @test @inferred(LittleDict(iter)) == LittleDict{Int,Float64}(2=>2.0, 3=>3.0)
+        
+        iter = Iterators.drop(1:10, 1)
+        @test_throws ArgumentError LittleDict(iter)
+
+        k_iter = Iterators.filter(x->x>1, [1,2,3,4])
+        v_iter = Iterators.filter(x->x>1, [1.0,2.0,3.0,4.0])
+        @test @inferred(LittleDict(k_iter, v_iter)) isa
+            LittleDict{Int,Float64, Vector{Int}, Vector{Float64}}
+
+        @test @inferred(LittleDict{Int, Char}(rand(1:100,20), rand('a':'z', 20))) isa
+            LittleDict{Int64,Char,Array{Int64,1},Array{Char,1}}
+    end
+    
+    @testset "empty dictionary" begin
+        d = LittleDict{Char, Int}()
+        @test length(d) == 0
+        @test isempty(d)
+        @test_throws KeyError d['c'] == 1
+        d['c'] = 1
+        @test !isempty(d)
+        @test_throws KeyError d[0.01]
+        @test isempty(empty(d))
+        empty!(d)
+        @test isempty(d)
+
+        # access, modification
+        for c in 'a':'z'
+            d[c] = c - 'a' + 1
+        end
+
+        @test (d['a'] += 1) == 2
+        @test 'a' in keys(d)
+        @test haskey(d, 'a')
+        @test get(d, 'B', 0) == 0
+        @test getkey(d, 'b', nothing) == 'b'
+        @test getkey(d, 'B', nothing) == nothing
+        @test !('B' in keys(d))
+        @test !haskey(d, 'B')
+        @test pop!(d, 'a') == 2
+
+        @test collect(keys(d)) == collect('b':'z')
+        @test collect(values(d)) == collect(2:26)
+        @test collect(d) == [Pair(a,i) for (a,i) in zip('b':'z', 2:26)]
+    end
+
+    @testset "convert" begin
+        d = LittleDict{Int,Float32}(i=>Float32(i) for i = 1:10)
+        @test convert(LittleDict{Int,Float32}, d) === d
+        dc = convert(LittleDict{Int,Float64}, d)
+        @test dc !== d
+        @test keytype(dc) == Int
+        @test valtype(dc) == Float64
+        @test keys(dc) == keys(d)
+        @test collect(values(dc)) == collect(values(d))
+    end
+
+    @testset "Issue #60" begin
+        od60 = LittleDict{Int,Int}()
+        od60[1] = 2
+
+        ranges = [2:5, 6:9, 10:13]
+        for range in ranges
+            for i = range
+                od60[i] = i+1
+            end
+            for i = range
+                delete!( od60, i )
+            end
+        end
+        od60[14]=15
+
+        @test od60[14] == 15
+    end
+
+
+    ##############################
+    # Copied and modified from Base/test/dict.jl
```

Comment:
Might be good to modify the `OrderedDict` tests to match these.  Maybe create an issue?

Reply:
No, the changes in this section are because you can't put `2000` elements in a `LittleDict`